baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    /**
+     * Scans the classpath for any migrations.json files and loads all defined {@link MigrationTarget} configurations.
+     */
+    protected void loadMigrations() {
+        Enumeration<URL> migrationsEnumeration = null;
+        try {
+            migrationsEnumeration = getClass().getClassLoader().getResources(migrationsFileName);
+
+        } catch (IOException ioe) {
+            throw new BatonException("Unable to find migrations!", ioe);
+        }
+
+        URL migrationsResource;
+        while (migrationsEnumeration.hasMoreElements()) {
+            migrationsResource = migrationsEnumeration.nextElement();
+            getLog().info(String.format("Loading migrations from: %s", migrationsResource.toString()));
+
+            try (InputStream migrationsStream = migrationsResource.openStream()) {
+                File tempMigrationsFile = File.createTempFile("migrations", ".json");
+                Files.copy(
+                        migrationsStream,
+                        tempMigrationsFile.toPath(),
+                        StandardCopyOption.REPLACE_EXISTING);
+                migrations = loadMigrationsJson(tempMigrationsFile, migrations);
+
+                getLog().info(String.format("Found %d migrations", migrations.size()));
+            } catch (IOException e) {
+                throw new BatonException("Unable to parse " + migrationsFileName, e);
+            }
+        }
+
+    }
+
+    /**
+     * Loads all {@link MigrationTarget}s contained within the given {@link InputStream}, which is expected to
+     * reference the desired migrations.json file to load.
+     *
+     * @param migrationsFile   {@link File} referencing migrations.json file desired to load.
+     * @param migrationTargets the migration targets already loaded to this point
+     * @return {@link Map} containing all loaded {@link MigrationTarget}s with their corresponding name as the map key.
+     */
+    protected Map<String, MigrationTarget> loadMigrationsJson(File migrationsFile,
+                                                              Map<String, MigrationTarget> migrationTargets) {
+
+        List<MigrationTarget> loadedMigrations = AbstractValidatedElement.readAndValidateJsonList(migrationsFile,
+                objectMapper, MigrationTarget.class);
+
+        for (MigrationTarget migrationTarget : loadedMigrations) {
+            migrationTargets.put(migrationTarget.getName(), migrationTarget);
+        }
+
+        return migrationTargets;
+    }
+
+    protected FileSet[] getFileSetsForTarget(MigrationTarget target) {
+        List<FileSet> localFileSets = new ArrayList<>();
+        for (org.technologybrewery.baton.config.FileSet targetFileSet : target.getFileSets()) {
+            FileSet fileSet = new FileSet();
+            fileSet.setDirectory(targetFileSet.getDirectory());
+            if (fileSet.getDirectory() == null) {
+                fileSet.setDirectory(baseDirectory.getAbsolutePath());
```

Comment:
Same note that it'd be nice to mention that migration directories trump plugin config.

Reply:
✅ completed